frame/support/procedural/src/benchmark.rs

```diff
@@ -199,6 +224,25 @@ impl BenchmarkDef {
 			params.push(ParamDef { name, typ: typ.clone(), start, end });
 		}
 
+		// ensure ReturnType is a Result<(), BenchmarkError>, if specified
+		if let ReturnType::Type(_, typ) = &item_fn.sig.output {
+			let non_unit = |span| return Err(Error::new(span, "expected `()`"));
+			// for this to parse as a ReturnType::Type, the contents must be a TypePath, QED
```

Comment:
Not exactly, it can be an `impl Trait` type, so we shouldn't panic here.

Reply:
But I think we don't want to allow anything but `Result<(), BenchmarkError>`?

Reply:
Yeah, an error message would suffice, but panicking would be very weird.

Reply:
ok added that and a UI test to cover it as well 👍🏻